cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -348,16 +350,15 @@ CUML_KERNEL void optimize_batch_kernel(T const* head_embedding,
  * @param rounding:    Floating rounding factor used to truncate the gradient update for
  *                     deterministic result.
  */
-template <typename T, int TPB_X>
+template <typename T, typename nnz_t, nnz_t TPB_X>
 void call_optimize_batch_kernel(T const* head_embedding,
                                 T* head_buffer,
-                                int head_n,
                                 T const* tail_embedding,
                                 T* tail_buffer,
-                                const MLCommon::FastIntDiv& tail_n,
+                                int tail_n,
```

Comment:
We shouldn't be casting to `int` here, should we?

Reply:
It is not casted since the calling function is now using an int. The `FastIntDiv` utility should only be useful for the CUDA kernels.